pkg/util/policy/agent/agent.py

```diff
@@ -395,22 +522,31 @@ def _make_u32_match(addr_scheme,
     """
     mask = src = dst = 0
 
+    # Take network bits from Romana CIDR. e.g. 10.0.0.0/8
+    #                                                   ^
+    network_width = int(addr_scheme["cidr"].split("/")[-1])
+
+    # Take first octet of Romana network e.g. 10.0.0.0/8
+    #                                         ^^
+    network_value = int(addr_scheme["cidr"].split(".")[0])
+
+
     # Full match on net portion.
-    mask = ((1<<addr_scheme['network_width'])-1) << 24
-    src  = addr_scheme['network_value'] << 24
-    dst  = addr_scheme['network_value'] << 24
+    mask = ((1<<network_width)-1) << 24
+    src  = network_value << 24
+    dst  = network_value << 24
 
     # Leaving the host portion empty...
 
     # Adding the mask and values for tenant
-    shift_by = addr_scheme['segment_width'] + addr_scheme['endpoint_width']
-    mask |= ((1<<addr_scheme['tenant_width'])-1) << shift_by
+    shift_by = addr_scheme['segment_bits'] + addr_scheme['endpoint_bits']
```

Comment:
We actually don't need to incorporate the tenant ID in these 'applied to' portion of the u32 match, since we already know that we are in a tenant-specific chain.